rule/time-equal.go

```diff
@@ -0,0 +1,70 @@
+package rule
+
+import (
+	"fmt"
+	"go/ast"
+	"go/token"
+
+	"github.com/mgechev/revive/lint"
+)
+
+// TimeEqualRule shows where "==" and "!=" used for equality check time.Time
+type TimeEqualRule struct{}
+
+// Apply applies the rule to given file.
+func (*TimeEqualRule) Apply(file *lint.File, _ lint.Arguments) []lint.Failure {
+	var failures []lint.Failure
+
+	onFailure := func(failure lint.Failure) {
+		failures = append(failures, failure)
+	}
+
+	w := &lintTimeEqual{file, onFailure}
+	w.file.Pkg.TypeCheck()
```

Comment:
`TypeCheck` might return an error, need to check for it

Reply:
Hi, thanks for the comments, I agree but saw that this error has not been checked in other files too.

Reply:
>this error has not been checked in other files too.
This is something that need to be fixed

Reply:
> > this error has not been checked in other files too.
> 
> This is something that need to be fixed

well, i thinks `return nil` is good , because the `Apply` method not returns any error.